### PR TITLE
Pull pre-build ex_doc from github for releases

### PR DESCRIPTION
**Pull request by starbelly**
This commit adjusts the release script to start pulling a pre-built ex-doc escript from elixir-lang's github
(i.e., https://github.com/elixir-lang/ex_doc/releases/tag/v0.31.0).

In addition, src, test, and mix.exs have been adjusted to reference and align with the escript name on ex-docs release artifiacts. Likewise, we also will now include the sha256sum in priv to be published along with the escript.

### CODE REVIEW

**Comment by paulo-ferraz-oliveira** [comment]
I can understand the "what" in this, which is the direction for my approval, but what's the "why"?

**Review by paulo-ferraz-oliveira — APPROVED** [review]

**Comment by starbelly** [comment]
> I can understand the "what" in this, which is the direction for my approval, but what's the "why"?

I suppose providing a synopsis on the why would have been useful here. 

History : Originally when this plugin was discussed it was my thought that the ex_doc maintainers build and publish an escript and that this plugin simply leverage said pre-built escript, I can not remember why they did not want to do this at the time.  I think also at the time, I thought that this plugin should download the escript vs publishing it. 

The why at this point in time is simply going with "cannon" if you will. It's arguable that it's probably also one less thing to have to do (mess up? 😄 ) as well. 

There are no strong technical reasons other than that though, we could of course continue to build and publish the escript ourselves, after all, the updates from our end won't come any quicker since it is still up to us to pull in and publish the latest. The only thing that really would make a difference is : Which approach is more secure? That I suppose boils down do you trust one of our machines more than a github CI runner or the other way around? 

Do you see any problems?

**Comment by paulo-ferraz-oliveira** [comment]
No problems, really. If it's just basically for security purposes, go for it.

File: release.sh [hunk 1]
```diff
@@ -44,11 +41,23 @@ cd rebar3_ex_doc
 
 git checkout "v${VERSION}"
 
-mix deps.get
-mix escript.build
+EX_DOC_VER="v$(grep -m1 "@ex_doc_version" mix.exs | awk '{print $2}' | tr -d '"')"
+echo "Fetching ex_doc ${EX_DOC_VER} escript from github ..."
```
Comment (paulo-ferraz-oliveira) [inline release.sh:45]:
```suggestion
echo "Fetching ex_doc ${EX_DOC_VER} escript from GitHub ..."
```

**Pull request merged**